application/src/main/java/org/togetherjava/tjbot/commands/reminder/ReminderCommand.java

```diff
@@ -65,14 +76,22 @@ public RemindCommand(Database database) {
                 "period to remind you in, the unit of time (e.g. 5 [weeks])", true);
         TIME_UNITS.forEach(unit -> timeUnit.addChoice(unit, unit));
 
-        getData().addOptions(timeUnit, timeAmount)
-            .addOption(OptionType.STRING, CONTENT_OPTION, "what to remind you about", true);
+        getData().addSubcommands(
+                new SubcommandData(CREATE_SUBCOMMAND, "creates a reminder").addOptions(timeAmount,
+                        timeUnit,
+                        new OptionData(OptionType.STRING, CONTENT_OPTION,
+                                "what to remind you about", true)),
+                new SubcommandData(LIST_SUBCOMMAND, "shows all your currently pending reminders"));
 
         this.database = database;
     }
 
     @Override
     public void onSlashCommand(SlashCommandInteractionEvent event) {
+        subcommandHandler.get(event.getSubcommandName()).accept(event);
+    }
+
+    private void handleAddCommand(SlashCommandInteractionEvent event) {
```

Comment:
Is having a map here really neccesairy? In my opinion it would be way more readable with a simple switch. You dont even use it anywhere else.

Reply:
with a map i can decrease the number of places one will need to change when introducing a new subcommand

Reply:
No. Its just one place anyways